charms/k8s/charmcraft.yaml

```diff
@@ -21,3 +59,20 @@ config:
 parts:
   charm:
     build-packages: [git]
+    charm-entrypoint: worker/src/charm.py
+  lib:
+    # move the ./worker/lib path to ./lib since 
+    # charmcraft assumes it to be there once the charm runs
+    after: [charm]
+    plugin: nil
+    source: ./
+    override-prime: |
+      mv worker/lib lib
+
+peers:
+  cluster:
+    interface: cluster
+
+provides:
+  tokens:
+    interface: tokens
```

Comment:
@mateoflorido if this isn't the `tokens` interface?  What is it.  So far as i can tell on the worker unit the relation name should be `cluster` but the interface it self can be anything.  

ideas for the interface name (not relation name)
* k8s-cluster -- this is how to cluster other components via the `k8s add-node/join-node` syntaxt
* k8s-worker -- this is a special relation to only `k8s-worker` nodes
* kube-control -- use the existing kube-control from charmed-kubernetes
    -- this would give a false impression we could use the existing "kubernetes-worker" charm with this charm

Reply:
It's the same tokens interface. We created tokens in Charmed Kubernetes for sharing the k8s authentication tokens over relation data (as we couldn't use Juju secrets). I like `k8s-cluster`. Why do we want to include `kube-control`? If I recall correctly, we need it for the cloud providers and maybe the new k8s-e2e charm. Should we create a new interface to avoid confusion and prevent people from trying to integrate incompatible charms?

Reply:
great. let's go with `k8s-cluster` for the interface name.  I'll change it here for now instead of being "tokens".  But won't provide an implementation for that relation